src/sleep/sleep.c

```diff
@@ -141,6 +146,38 @@ static int write_state(FILE **f, char **states) {
         return r;
 }
 
+static int configure_hibernation(void) {
+        _cleanup_free_ char *resume = NULL, *resume_offset = NULL;
+        int r;
+
+        /* check for proper hibernation configuration */
+        r = read_one_line_file("/sys/power/resume", &resume);
+        if (r < 0)
+                return log_debug_errno(r, "Error reading from /sys/power/resume: %m");
+
+        r = read_one_line_file("/sys/power/resume_offset", &resume_offset);
+        if (r < 0)
+                return log_debug_errno(r, "Error reading from /sys/power/resume_offset: %m");
+
+        if ((resume_offset && !streq(resume_offset, "0")) && (resume && !streq(resume, "0:0"))) {
```

Comment:
given that on failure to read the two files you exit early from this function it's guaranteed that the two variables are non-NULL when you reach this, hence the explicit checks for non-NULL-ness can be dropped.